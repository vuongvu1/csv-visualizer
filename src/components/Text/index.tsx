import React from "react";
import SC from "./styles";

export enum TextType {
  H1 = "h1",
  H2 = "h2",
  BODY = "body",
}

interface Props {
  type?: TextType;
}

const Text: React.FC<Props> = ({ type, children }) => {
  return (
    <SC.Container as={type || "div"} type={type}>
      {children}
    </SC.Container>
  );
};

export default Text;
