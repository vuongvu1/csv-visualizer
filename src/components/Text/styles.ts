import styled, { css } from "styled-components";
import {
  primaryColor,
  secondaryColor,
  neutralColor,
  blackColor,
} from "configs/colors";
import { FontSize } from "configs/styles";
import { TextType } from "./index";

export default {
  Container: styled.div<{ type?: TextType }>`
    color: ${neutralColor};
    font-weight: normal;
    font-size: ${FontSize.medium};

    ${({ type }) => {
      if (type === TextType.H1) {
        return css`
          color: ${primaryColor};
          font-weight: bold;
          font-size: ${FontSize.large};
        `;
      }
      if (type === TextType.H2) {
        return css`
          color: ${secondaryColor};
          font-weight: bold;
          font-size: ${FontSize.large};
        `;
      }

      if (type === TextType.BODY) {
        return css`
          color: ${blackColor};
          font-weight: normal;
          font-size: ${FontSize.medium};
        `;
      }
    }}
  `,
};
