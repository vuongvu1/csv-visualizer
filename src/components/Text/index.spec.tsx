import React from "react";
import { render, cleanup } from "@testing-library/react";
import Text, { TextType } from "./index";

describe("Text", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const TextH1 = render(<Text type={TextType.H1} />);
    const TextH2 = render(<Text type={TextType.H2} />);
    const TextBody = render(<Text type={TextType.BODY} />);

    expect(TextH1.container).toMatchSnapshot();
    expect(TextH2.container).toMatchSnapshot();
    expect(TextBody.container).toMatchSnapshot();
  });
});
