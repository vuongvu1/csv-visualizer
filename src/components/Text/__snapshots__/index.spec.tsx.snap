// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
.c0 {
  color: #EFEFEF;
  font-weight: normal;
  font-size: 16px;
  color: #3F7EF0;
  font-weight: bold;
  font-size: 32px;
}

<div>
  <h1
    class="c0"
    type="h1"
  />
</div>
`;

exports[`Text should match snapshot 2`] = `
.c0 {
  color: #EFEFEF;
  font-weight: normal;
  font-size: 16px;
  color: #63AEF7;
  font-weight: bold;
  font-size: 32px;
}

<div>
  <h2
    class="c0"
    type="h2"
  />
</div>
`;

exports[`Text should match snapshot 3`] = `
.c0 {
  color: #EFEFEF;
  font-weight: normal;
  font-size: 16px;
  margin: 0;
  color: #000000;
  font-weight: normal;
  font-size: 16px;
}

<div>
  <p
    class="c0"
    type="p"
  />
</div>
`;
